src/components/Semester/Semester.vue

```diff
@@ -406,10 +374,12 @@ export default defineComponent({
       this.isConfirmationOpen = false;
     },
     // TODO @willespencer refactor the below methods after gatekeep removed (to only 1 method)
-    addCourse(data: CornellCourseRosterCourse, choice: FirestoreCourseOptInOptOutChoices) {
+    addCourse(data: CornellCourseRosterCourse, selectableReqId: string) {
       const newCourse = cornellCourseRosterCourseToFirebaseSemesterCourseWithGlobalData(data);
-      // Since the course is new, we know the old choice does not exist.
-      addCourseToSemester(this.year, this.season, newCourse, () => choice, this.$gtag);
+      if (selectableReqId) {
+        addAcknowledgedCheckerWarningOptIn(newCourse.uniqueID, selectableReqId);
+      }
+      addCourseToSemester(this.year, this.season, newCourse, this.$gtag);
```

Comment:
I'm honestly not sure why `selectableReqId`, from the child `NewCourseModal.vue`, was not taken into consideration here. I think this should be tested a bit.

To maintain the old behavior, the call to `addAcknowledgedCheckerWarningOptIn` should be unnecessary. But I'm not sure that's correct.